### PR TITLE
[-] FO : make the order available in order confirmation step

### PR DESCRIPTION
**Pull request by mickaelandrieu**
@nihco2 note you can already use `$is_guest` to check if you need to include a subscription form.

Now `order` is available, you can do `$url_to_invoice` to get the invoice url (the availability depends on many factors ).

Let me know if you need more :)

### CODE REVIEW

File: controllers/front/OrderConfirmationController.php [hunk 1]
```diff
@@ -81,11 +81,14 @@ public function init()
...
         parent::initContent();
+        $order = new Order(Order::getOrderByCartId((int)($this->id_cart)));
+        $order->url_to_invoice = HistoryController::getUrlToInvoice($order, $this->context);
 
         $this->context->smarty->assign(array(
             'is_guest' => $this->context->customer->is_guest,
             'HOOK_ORDER_CONFIRMATION' => $this->displayOrderConfirmation(),
-            'HOOK_PAYMENT_RETURN' => $this->displayPaymentReturn()
+            'HOOK_PAYMENT_RETURN' => $this->displayPaymentReturn(),
+            'order' => new Order(Order::getOrderByCartId((int)($this->id_cart)))
```
Comment (djfm) [inline controllers/front/OrderConfirmationController.php]: I think you already have your `$order` 5 lines above

Comment (mickaelandrieu) [inline controllers/front/OrderConfirmationController.php]: nice catch ;)

File: controllers/front/OrderConfirmationController.php [hunk 2]
```diff
@@ -81,11 +81,14 @@ public function init()
...
         parent::initContent();
+        $order = new Order(Order::getOrderByCartId((int)($this->id_cart)));
+        $order->url_to_invoice = HistoryController::getUrlToInvoice($order, $this->context);
 
         $this->context->smarty->assign(array(
             'is_guest' => $this->context->customer->is_guest,
             'HOOK_ORDER_CONFIRMATION' => $this->displayOrderConfirmation(),
-            'HOOK_PAYMENT_RETURN' => $this->displayPaymentReturn()
+            'HOOK_PAYMENT_RETURN' => $this->displayPaymentReturn(),
+            'order' => $order
```
Comment (djfm) [inline controllers/front/OrderConfirmationController.php]: Please don't pass `ObjectModel`'s to templates, as really dangerous actions could be performed on such an object, especially an order.

Please either serialize the whole thing as array (see `Adapter\ObjectSerializer`) or choose the properties you need in the template yourself.

Comment (mickaelandrieu) [inline controllers/front/OrderConfirmationController.php]: Agree, improved my work regarding your comment.

**Comment by mickaelandrieu** [comment]
ping  @djfm I don't understand why Travis don't run a build for this contribution ... but in my own account the build is green => https://travis-ci.org/mickaelandrieu/PrestaShop/builds/108328013

File: controllers/front/OrderConfirmationController.php [hunk 3]
```diff
@@ -81,11 +81,14 @@ public function init()
...
         parent::initContent();
+        $order = new Order(Order::getOrderByCartId((int)($this->id_cart)));
 
         $this->context->smarty->assign(array(
             'is_guest' => $this->context->customer->is_guest,
             'HOOK_ORDER_CONFIRMATION' => $this->displayOrderConfirmation(),
-            'HOOK_PAYMENT_RETURN' => $this->displayPaymentReturn()
+            'HOOK_PAYMENT_RETURN' => $this->displayPaymentReturn(),
+            'url_to_invoice' => HistoryController::getUrlToInvoice($order, $this->context),
+            'products' => $order->getCartProducts()
```
Comment (djfm) [inline controllers/front/OrderConfirmationController.php]: Please remove this for now. We need the whole order but we don't have an easy way to present it yet.

Comment (mickaelandrieu) [inline controllers/front/OrderConfirmationController.php]: Done, thank you for review ;)

**Pull request merged**